### ISSUE TITLE
Update without restarting the station

### ISSUE THREAD

**Issue by fafafoya** [AzuraCast/AzuraCast#1840]
Looking for a solution to update Azuracast using Docker without restarting the stations/platform.
When listeners are streaming, it is a bad point for the station.

**Comment by SlvrEagle23** [comment]
@fafafoya There are a lot of technical reasons this isn't currently possible with our Docker installation. All of our containers depend heavily on one another and communicate back and forth, and shutting down one means the others may be able to remain up, but would fail to operate in important ways. We also have to remove some volumes (the temporary storage) during the update process, which requires that all of the containers be shut down. Also, currently there is no way for Docker to report back to us which containers are updated, so that we can avoid shutting down the ones that haven't changed at all.

Basically...at the moment it's not possible. We've done all we can to reduce the update time (caching many of the update components, making the "pull new images" step happen with the services still live, etc) but the update isn't a zero-downtime process and probably can't be for a while.
